core/domain/acl_decorators.py

```diff
@@ -126,3 +171,263 @@ def test_can_view_stats(self, exploration_id, **kwargs):
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to access creator dashboard.')
+
+    return test_can_access
+
+
+def can_view_exploration_feedback(handler):
+    """Decorator to check whether the user can view feedback for a given
+    exploration.
+    """
+
+    def test_can_access(self, exploration_id, **kwargs):
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_id):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to view feedbacks.')
+
+    return test_can_access
+
+
+def can_rate_exploration(handler):
+    """Decorator to check whether the user can give rating to given
+    exploration.
+    """
+
+    def test_can_rate(self, exploration_id, **kwargs):
+        if role_services.ACTION_RATE_EXPLORATION in self.actions:
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to give ratings to explorations.')
+
+    return test_can_rate
+
+
+def can_flag_exploration(handler):
+    """Decorator to check whether user can flag given exploration."""
+
+    def test_can_flag(self, exploration_id, **kwargs):
+        if role_services.ACTION_FLAG_EXPLORATION in self.actions:
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to flag explorations.')
+
+    return test_can_flag
+
+
+def can_subscribe_to_users(handler):
+    """Decorator to check whether user can subscribe/unsubscribe a creator."""
+
+    def test_can_subscribe(self, **kwargs):
+        if role_services.ACTION_SUBSCRIBE_TO_USERS in self.actions:
+            return handler(self, **kwargs)
+        else:
+            raise base.UserFacingExceptions.UnauthorizedUserException(
+                'You do not have credentials to manage subscriptions.')
+
+    return test_can_subscribe
+
+
+def can_edit_exploration(handler):
+    """Decorator to check whether the user can edit given exploration."""
+
+    def _is_state_valid(escaped_state_name, state_name_list):
+        if escaped_state_name is None:
+            return True
+        state_name = utils.unescape_encoded_uri_component(
+            escaped_state_name)
+        return bool(state_name in state_name_list)
+
+    def test_can_edit(self, exploration_id, *args, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_id):
+            exploration = exp_services.get_exploration_by_id(exploration_id)
+            escaped_state_name = (
```

Comment:
Ah, I think the only handler (`ResolvedAnswersHandler`) that depended on the escaped_state_name variable has been removed as a part of one of the above comments.

So, I can remove this part from decorator anyways.